### PR TITLE
NO-ISSUE: Add JSON schema for the service Helm chart

### PR DESCRIPTION
**Pull request by jhernand**
## Summary

- Add a `values.schema.json` file to validate the service chart's `values.yaml` at lint and
  install time, enforcing types, enums, and structure for all values including the reusable
  ConfigMap/Secret credential source pattern.
- Default `auth.controllerCredentials` and `idp.credentials` to empty arrays instead of null,
  and guard the controller deployment template's `range` blocks with `if` checks so empty
  credential lists render valid YAML.
- Add a Ginkgo test in `internal/chart` that runs `helm lint` against the service chart with
  all required values populated, catching schema and template regressions automatically.

## Test plan

- [x] `ginkgo run -v internal/chart` passes (helm lint succeeds with valid values).
- [x] Verify `helm lint charts/service/` with real deployment values still works.
- [x] Verify `helm install --dry-run` renders correct manifests.

## Summary by CodeRabbit

* **New Features**
  * Added JSON Schema validation for Helm chart values to enforce configuration structure.

* **Improvements**
  * Safer conditional rendering for credential volumes in deployment templates.
  * Default credential lists set to empty arrays.

* **Tests**
  * Added automated Helm lint/validation tests for the chart.

* **Chores**
  * CI updated to install a pinned Helm tool before relevant jobs.

### CODE REVIEW

File: charts/service/values.schema.json [hunk 1]
```diff
@@ -0,0 +1,282 @@
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
```
[the remark this answers is not in the document]
Comment (jhernand) [reply charts/service/values.schema.json:2]: Will add a test to ensure that Helm is at least version 4.

File: internal/chart/chart_suite_test.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	Expect(err).ToNot(HaveOccurred())
+
+	// Check that it is indeed the chart directory:
+	chartFile = filepath.Join(chartDir, "Chart.yaml")
+	_, err = os.Stat(chartFile)
+	Expect(err).ToNot(
+		HaveOccurred(),
+		"Chart file '%s' doesn't exist",
+		chartDir,
+	)
```
[the remark this answers is not in the document]
Comment (jhernand) [reply internal/chart/chart_suite_test.go:103]: Done.

**Review by tzvatot — COMMENTED** [review]
## Review

Well-structured PR — the JSON schema is thorough, the `$defs` pattern for credential sources avoids repetition, and the helm lint test catches regressions automatically. Two CodeRabbit findings already addressed (error message arg, Helm 4 version gate).

| Category | Count |
|----------|-------|
| 🔴 Critical | 0 |
| 🟡 Important | 1 |
| 💡 Suggestion | 0 |

---

### 🟡 Empty credentials case not tested — `else []` may render invalid YAML

**File:** `charts/service/templates/controller/deployment.yaml:68-70` and `internal/chart/chart_lint_test.go`

The `{{ else }} [] {{ end }}` branch handles empty credential lists, but the helm lint test always provides non-empty credentials. Without whitespace trimming (`{{-`), the rendered output for empty credentials would be:

```yaml
      sources:
      
          []
      
```

The blank lines between `sources:` and `[]` may cause YAML parsing issues (blank lines can terminate a mapping value in some parsers). CodeRabbit flagged this too.

**Recommendation:** Add a second lint test case with empty credentials (`controllerCredentials: []`, `credentials: []`) to verify the template renders valid YAML in that path. If it fails, switch to `{{- else }} [] {{- end }}` to trim the whitespace.

---

**Overall:** LGTM pending the empty-credentials test. The schema is comprehensive, the CI integration is clean, and the credential source `$defs` pattern is well-designed for reuse.

File: internal/chart/chart_suite_test.go [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+var _ = BeforeSuite(func() {
+	var err error
+
+	// Create the logger:
+	logger, err = logging.NewLogger().
+		SetLevel(slog.LevelDebug.String()).
+		SetWriter(GinkgoWriter).
+		Build()
+	Expect(err).ToNot(HaveOccurred())
```
[the remark this answers is not in the document]
Comment (jhernand) [reply internal/chart/chart_suite_test.go:58]: That guidance doesn't apply for these tests: they just test structure of the Helm charts, and there is no authentication or database needed for that.

File: internal/chart/chart_suite_test.go [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+		Stdout: io.MultiWriter(helmOut, GinkgoWriter),
+		Stderr: GinkgoWriter,
+	}
+	err = helmCmd.Run()
+	Expect(err).ToNot(HaveOccurred())
+	helmVersion := strings.TrimSpace(helmOut.String())
+	Expect(helmVersion).To(
+		MatchRegexp("^v4\\."),
+		"Helm version is '%s' but must be at least version 4",
+		helmVersion,
```
[the remark this answers is not in the document]
Comment (jhernand) [reply internal/chart/chart_suite_test.go:85]: Won't do.
